cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,392 @@
+        },
+        re.compile(r'graph\s*?='): {
+            'short': (
+                '``[cycle point]graph =`` is deprecated, '
+                'use ``cycle point = <graph>``'
+            ),
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(SECTION2.format('remote')): {
+            'short': (
+                '``[runtime][<namespace>][remote]host`` is deprecated, '
+                'use ``[runtime][<namespace>]platform``'
+            ),
+            'url': 'major-changes/platforms.html#platforms'
+        },
+        re.compile(SECTION3.format('job')): {
+            'short': (
+                '``[runtime][<namespace>][job]`` is deprecated, '
+                'use ``[runtime][<namespace>]platform``'
+            ),
+            'url': 'major-changes/platforms.html#platforms'
+        },
+        re.compile(SECTION2.format('parameter templates')): {
+            'short': (
+                '``[cylc][parameter templates]`` is deprecated, '
+                'use ``[task parameters][templates]``'
+            ),
+            'url': ''
+        },
+        re.compile(SECTION2.format('parameters')): {
+            'short': (
+                '``[cylc][parameters]`` is deprecated, '
+                'use ``[task parameters]``'
+            ),
+            'url': ''
+        },
+        re.compile(r'task event mail interval\s*?='): {
+            'short': (
+                '``[cylc][task event mail interval]`` is deprecated, '
+                'use ``[scheduler][mail][task event batch interval]``'
+            ),
+            'url': ''
+        }
+    },
+    'lint': {
+        re.compile(r'^\t'): {
+            'short': 'Use multiple spaces, not tabs',
+            'url': STYLE_GUIDE + 'tab-characters'
+        },
+        # Not a full test, but if a non section is not indented...
+        re.compile(r'^[^\[|\s]'): {
+            'short': 'Item not indented.',
+            'url': STYLE_GUIDE + 'indentation'
+        },
+        #            [section]
+        re.compile(r'^\s+\[.*\]'): {
+            'short': 'Too many indents for top level section.',
+            'url': STYLE_GUIDE + 'indentation'
+        },
+        # 2 or 4 space indentation both seem reasonable:
+        re.compile(r'^(\s|\s{3}|\s{5,})\[\[.*\]\]'): {
+            'short': 'wrong number of indents for second level section.',
+            'url': STYLE_GUIDE + 'indentation'
+        },
+        re.compile(r'^(\s{1,3}|\s{5,7}|\s{9,})\[\[\[.*\]\]\]'): {
+            'short': 'wrong number of indents for third level section.',
+            'url': STYLE_GUIDE + 'indentation'
+        },
+        re.compile(r'\s$'): {
+            'short': 'wrong number of indents for third level section.',
+            'url': STYLE_GUIDE + 'trailing-whitespace'
+        },
+        re.compile(r'^.{80,}'): {
+            'short': 'line > 79 characters.',
+            'url': STYLE_GUIDE + 'line-length-and-continuation'
+        },
```

Comment:
I feel like this is very much down to opinion. If there is no way to turn it off then we should probably leave it out

Reply:
I've commented it, I'm not going to prioritize the linting side of things.